src/formatDate.js

```diff
@@ -8,7 +8,54 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [oldData, idFromYear, oldYear, fromSumbol] = getFormat(fromFormat);
+  const [newData, , newYear, toSumbol] = getFormat(toFormat);
+
+  const fromDate = date.split(fromSumbol);
+  const toDate = [];
+
+  const year = fromDate[idFromYear];
+
+  let formatYear = '';
+
+  if (oldYear !== newYear) {
+    if (newYear < oldYear) {
+      formatYear += year.slice(2);
+    }
+
+    if (newYear > oldYear) {
+      if (year >= 30) {
+        formatYear += '19' + year;
+      } else {
+        formatYear += '20' + year;
+      }
+    }
+
+    fromDate[idFromYear] = formatYear;
+  }
+
+  oldData[idFromYear] = newYear;
+
+  for (const format of newData) {
+    const index = oldData.indexOf(format);
+
+    toDate.push(fromDate[index]);
+  }
+
+  return toDate.join(toSumbol);
+}
+
+function getFormat(format) {
+  const [first, second, therd, sumbol] = format;
+  const formatData = [first, second, therd];
+
+  const idYear = formatData.includes('YYYY')
```

Comment:
The logic for determining `idYear` assumes that 'YYYY' or 'YY' will always be present. Ensure that the input format always contains a year component to avoid potential errors.